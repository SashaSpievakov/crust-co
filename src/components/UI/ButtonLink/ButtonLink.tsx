import { FC, PropsWithChildren } from 'react';
import { Link } from 'react-router-dom';

import { ButtonSecondary } from '../../../styles/Buttons.styled';

interface ButtonLinkProps extends PropsWithChildren {
  link: string;
}

export const ButtonLink: FC<ButtonLinkProps> = ({ link, children }) => {
  return (
    <ButtonSecondary to={link} as={Link}>
      {children}
    </ButtonSecondary>
  );
};
