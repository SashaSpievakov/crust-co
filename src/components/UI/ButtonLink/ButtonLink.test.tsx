import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {
  rendererWithProviders,
  renderWithProvidersAndRoutes,
} from '../../../tests/helpers';
import { ButtonLink } from './ButtonLink';

describe('ButtonBackToHome tests', () => {
  test('should render the ButtonBackToHome UI component', () => {
    const snapshot = rendererWithProviders(
      <ButtonLink link="/">Go Back</ButtonLink>,
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('should redirect to the Home page', () => {
    renderWithProvidersAndRoutes(null, true, '/testing');
    const link = screen.getByRole('link', {
      name: /go back/i,
    });

    userEvent.click(link);
    expect(screen.getByText(/all pizzas/i)).toBeInTheDocument();
  });
});
