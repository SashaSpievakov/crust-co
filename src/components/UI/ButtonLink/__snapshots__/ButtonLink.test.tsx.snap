// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBackToHome Tests renders the ButtonBackToHome UI component 1`] = `
Object {
  "children": Array [
    "Go Back",
  ],
  "props": Object {
    "className": "sc-hLBbgP sc-eDvSVe fGqcTx lhIsum",
    "href": "/",
    "onClick": [Function],
    "target": undefined,
  },
  "type": "a",
}
`;
